src/curl.cpp

```diff
@@ -1949,10 +1949,13 @@ bool S3fsCurl::ResetHandle(AutoLock::Type locktype)
     if(CURLE_OK != curl_easy_setopt(hCurl, CURLOPT_NOPROGRESS, 0)){
         return false;
     }
-    // [NOTE]
-    // CURLOPT_PROGRESSFUNCTION should be changed to CURLOPT_XFERINFOFUNCTION,
-    // but CnetOS7's curl version is old, so it haven't been changed yet.
-    if(CURLE_OK != curl_easy_setopt(hCurl, CURLOPT_PROGRESSFUNCTION, S3fsCurl::CurlProgress)){
+    if(CURLE_OK != curl_easy_setopt(hCurl,
```

Comment:
I would like to avoid defining this ifdef statement inline.
```
#if LIBCURL_VERSION_NUM >= 0x073100
         CURLOPT_XFERINFOFUNCTION
#else
         CURLOPT_PROGRESSFUNCTION
#endif
```

How about defining another symbol at the beginning of the file as shown below and using that symbol?
```
#if LIBCURL_VERSION_NUM >= 0x073100
#define S3FS_CURLOPT_XFERINFOFUNCTION   CURLOPT_XFERINFOFUNCTION
#else
#define S3FS_CURLOPT_XFERINFOFUNCTION   CURLOPT_PROGRESSFUNCTION
#endif
```
I prefer this one.

Reply:
Done.